src/scmdata/run.py

```diff
@@ -1257,17 +1252,27 @@ def interpolate(
 
         target_times = np.asarray(target_times, dtype="datetime64[s]")
 
-        res = self.copy(copy_ts=False)
+        res = self.copy()
+
+        target_times = TimePoints(target_times)
+
+        timeseries_converter = TimeseriesConverter(
+            self.time_points.values,
+            target_times.values,
+            interpolation_type=interpolation_type,
+            extrapolation_type=extrapolation_type,
+        )
+
+        target_data = np.zeros((len(target_times), len(res)))
 
-        res._ts = [
-            ts.interpolate(
-                target_times,
-                interpolation_type=interpolation_type,
-                extrapolation_type=extrapolation_type,
+        for i in range(len(res)):
```

Comment:
hmmm there must be a quicker way to do this, will require some thinking I suspect. Maybe one to mark with a todo?

Reply:
Yeah the interpolator should be changed to take 2d arrays, but now we are 20x faster according to the benchmarks.